File: loop_time_sample.go
package main

// Import required packages
import (
	"encoding/csv"
	"fmt"
	"log"
	"os"
	"time"
)

// Define main function
func main() {

	// Open file
	f, err := os.Open("data/full_matrix.csv")
	if err != nil {
		log.Fatal(err)
	}

	// Close the file at the end of the program
	defer f.Close()

	// Read csv into 2-d array
	csvReader := csv.NewReader(f)
	arr, err := csvReader.ReadAll()
	if err != nil {
		log.Fatal(err)
	}

	// Calculate loop start time
	var start_time = time.Now()

	// Accessing each value of the array
	fmt.Println("Elements of my array")
	for p := 0; p < len(arr); p++ {
		for q := 0; q < len(arr[0]); q++ {
			fmt.Println(arr[p][q])
		}
	}

	// Calculate loop end time
	var end_time = time.Now()

	// Print time to run
	fmt.Println("Time to run loop:", end_time.Sub(start_time))

}

File: loop_time_sample.py
# Load required packages
import numpy as np
import time

# Load array data
arr = np.loadtxt(open("data/full_matrix.csv", "rb"), delimiter=",", skiprows=1)

# Calcualte loop start time
start_time = time.time()

# Loop through array
for i in range(len(arr)):
    for j in range(len(arr[0])):
        print(arr[i][j])

# Calculate total time to run loop 
print(f"--- {(round(time.time() - start_time, 3))} seconds to complete array loop ---")
